python/pyarrow/pandas_compat.py

```diff
@@ -591,8 +590,7 @@ def convert_column(col, field):
     pandas_metadata = construct_metadata(df, column_names, index_columns,
                                          index_descriptors, preserve_index,
                                          types)
-    metadata = deepcopy(schema.metadata) if schema.metadata else dict()
-    metadata.update(pandas_metadata)
+    metadata = pa.KeyValueMetadata(schema.metadata or {}, **pandas_metadata)
```

Comment:
This change causes metadata to have a duplicate "pandas" field, while before, the `update` method would overwrite existing keys with the ones in the passed dict.